api/src/main/java/jakarta/data/metamodel/restrict/CompositeRestrictionRecord.java

```diff
@@ -38,10 +40,12 @@ record CompositeRestrictionRecord<T>(
             throw new IllegalArgumentException(
                     "Cannot create a composite restriction without any restrictions to combine.");
         }
+        Objects.requireNonNull(type, "type must not be null");
+        restrictions.forEach(r -> Objects.requireNonNull(r, "Restriction must not be null"));
```

Comment:
I believe that the restrictions `List` should not have a null item.

Reply:
Agreed.